aptos-move/framework/aptos-framework/sources/dora_committee.move

```diff
@@ -0,0 +1,1048 @@
+        id: u64,
+        node_address: address
+    ): NodeData acquires CommitteeInfoStore {
+        let committee_store = borrow_global<CommitteeInfoStore>(com_store_addr);
+        let committee = simple_map::borrow(&committee_store.committee_map, &id);
+        let (addrs, dora_nodes) = simple_map::to_vec_pair(committee.map);
+        let (flag, index) = vector::index_of(&addrs, &node_address);
+        let dora_node_info = vector::borrow(&dora_nodes, index);
+        assert!(flag, error::invalid_argument(NODE_NOT_FOUND));
+        NodeData {
+            operator: node_address,
+            ip_public_address: dora_node_info.ip_public_address,
+            dora_public_key: dora_node_info.dora_public_key,
+            network_public_key: dora_node_info.network_public_key,
+            elgamal_pub_key: dora_node_info.elgamal_pub_key,
+            network_port: dora_node_info.network_port,
+            rpc_port: dora_node_info.rpc_port,
+        }
+    }
+
+    #[view]
+    /// Get the committee's id for a single node
+    public fun get_committee_id_for_node(
+        com_store_addr: address,
+        node_address: address
+    ): u64 acquires CommitteeInfoStore {
+        let committee_store = borrow_global<CommitteeInfoStore>(com_store_addr);
+        let id = *simple_map::borrow(&committee_store.node_to_committee_map, &node_address);
+        id
+    }
+
+    #[view]
+    /// Get the dora node peers vector for a single node
+    public fun get_peers_for_node(
+        com_store_addr: address,
+        node_address: address
+    ): vector<NodeData> acquires CommitteeInfoStore {
+        let committee_id = get_committee_id_for_node(com_store_addr, node_address);
+        let this_node = get_node_info(com_store_addr, committee_id, node_address);
+        let dora_node_info = get_committee_info(com_store_addr, committee_id);
+        let (_, index) = vector::index_of(&dora_node_info, &this_node);
+        vector::remove(&mut dora_node_info, index);
+        dora_node_info
+    }
+
+    #[view]
+    /// Check if the committee has a valid dkg
+    public fun does_com_have_dkg(com_store_addr: address, com_id: u64): bool acquires CommitteeInfoStore {
+        let committee_store = borrow_global<CommitteeInfoStore>(com_store_addr);
+        let committee = simple_map::borrow(&committee_store.committee_map, &com_id);
+        committee.has_valid_dkg
+    }
+
+    /// Update the dkg flag
+    public fun update_dkg_flag(
+        com_store_addr: address,
+        owner_signer: &signer,
+        com_id: u64,
+        flag_value: bool
+    ) acquires CommitteeInfoStore {
+        // Only the OwnerCap capability can access it
+        let _acquire = &capability::acquire(owner_signer, &OwnerCap {});
+        let committee_store = borrow_global_mut<CommitteeInfoStore>(com_store_addr);
+        let committee = simple_map::borrow_mut(&mut committee_store.committee_map, &com_id);
+        committee.has_valid_dkg = flag_value;
```

Comment:
@dhaval-supraoracles , @nizam-supraoracles should we have an event for update of `DKG` flag as well?

Reply:
Resolved as finished.